src/vm/syncblk.inl

```diff
@@ -178,7 +178,7 @@ FORCEINLINE AwareLock::LeaveHelperAction AwareLock::LeaveHelper(Thread* pCurThre
     _ASSERTE((size_t)m_MonitorHeld & 1);
     _ASSERTE(m_Recursion >= 1);
 
-#if defined(_DEBUG) && defined(TRACK_SYNC)
+#if defined(_DEBUG) && defined(TRACK_SYNC) && !defined(CROSSGEN_COMPILE)
```

Comment:
@janvorli ``Thread`` does not have ``m_pTrackSync`` when ``CROSSGEN_COMPILE`` is set (please refer to the definition of ``Thread`` at the beginning of ``threads.h`` for details).

So, setting ``TRACK_SYNC`` in crossgen.cmake can't address this issue. Do we need to introduce ``m_pTrackSync`` in ``Thread`` for crossgen?